### ISSUE TITLE
Dnsmasq on OpenWrt 22.03 does not like conf-dir line in /etc/dnsmasq.conf

### ISSUE THREAD

**Issue by ilario** [libremesh/lime-packages#970]
A line gets added in /etc/dnsmasq.conf here:
https://github.com/libremesh/lime-packages/blob/15aa6d2636f9dc506ad7161cd13eca4ea0056a44/packages/lime-system/files/usr/lib/lua/lime/network.lua#L161

And its presence somehow causes that dnsmasq does not start on OpenWrt 22.03.
The problem is caused by the usage of procd-ujail and should be possible to solve like this:
https://github.com/openwrt/openwrt/issues/9726#issuecomment-1198828327

**Comment by ilario** [comment]
procd-ujail has been included by default since OpenWrt 22.03 ( https://github.com/openwrt/openwrt/commit/44f694ba1bca1417d24e851c637c284f9f78c06d ), so this issue should not be visible in previous OpenWrt releases.

Does anyone remember whether dnsmasq has to access other files anywhere? There are some files in the /tmp folder that maybe are going to be used and cause the same problem?

**Comment by ilario** [comment]
To be checked: I suspect that also this file will cause troubles
https://github.com/libremesh/lime-packages/blob/06399e5de7fb7b9d3cf13c19f7bf70accb11299a/packages/shared-state-dnsmasq_leases/files/etc/uci-defaults/90_dnsmasq-lease-share#L6

But, if this gets confirmed, it could be a bug on OpenWrt side, as the dhcphostsfile gets mentioned in the dnsmasq init script here:
https://github.com/openwrt/openwrt/blob/openwrt-22.03/package/network/services/dnsmasq/files/dnsmasq.init#L1024-L1025

But does not get included in the allowed ujail files here:
https://github.com/openwrt/openwrt/blob/openwrt-22.03/package/network/services/dnsmasq/files/dnsmasq.init#L1172-L1180

I repeat: this is not yet confirmed on a router.

**Comment by ilario** [comment]
Also to check:
the newly merged package "shared-state-dnsmasq_servers" #812 has a line that looks like it could be problematic:
https://github.com/libremesh/lime-packages/blob/c83cb1ebe2f09aec6f4b393e73c8b92b95668018/packages/shared-state-dnsmasq_servers/files/etc/uci-defaults/shared-state-dnsmasq_servers#L4

**Comment by ilario** [comment]
Both the two suspected files mentioned in the comments above are inaccessible to dnsmasq due to ujail:

```
# logread
[...]
daemon.err dnsmasq[1]: cannot read /tmp/dhcp.hosts_remote: No such file or directory
daemon.err dnsmasq[1]: cannot read /var/shared-state/dnsmasq_servers: No such file or directory
[...]
# ls -l /tmp/dhcp.hosts_remote
-rw-r--r--    1 root     root            29 Apr  5 09:39 /tmp/dhcp.hosts_remote
# ls -l /var/shared-state/dnsmasq_servers
-rw-r--r--    1 root     root             1 Apr  5 09:39 /var/shared-state/dnsmasq_servers
```

**Comment by ilario** [comment]
> To be checked: I suspect that also this file will cause troubles
> 
> https://github.com/libremesh/lime-packages/blob/06399e5de7fb7b9d3cf13c19f7bf70accb11299a/packages/shared-state-dnsmasq_leases/files/etc/uci-defaults/90_dnsmasq-lease-share#L6
> 
> But, if this gets confirmed, it could be a bug on OpenWrt side, as the dhcphostsfile gets mentioned in the dnsmasq init script here: https://github.com/openwrt/openwrt/blob/openwrt-22.03/package/network/services/dnsmasq/files/dnsmasq.init#L1024-L1025


And indeed it was a bug on the OpenWrt side, fixed in the openwrt-22.03 branch:
https://github.com/openwrt/openwrt/commit/936df715de3d33947ce38ca232b05c2bd3ef58f1

**Comment by ilario** [comment]
Closing, as OpenWrt 22.03.4 is out and as #1004 has been fixed
